src/features/messages/cards/TransactionCard.tsx

```diff
@@ -207,10 +207,23 @@ function DeliveryStatus({ children }: PropsWithChildren<unknown>) {
   );
 }
 
-function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
+function CallDataModal({ debugResult,chainId,message}: { debugResult?: MessageDebugResult,chainId:ChainId,message:Message }) {
   const [isOpen, setIsOpen] = useState(false);
+  const [disabled,setDisabled]=useState(false)
+  const [buttonText,setButtonText]=useState("Simulate call with Tenderly")
+  const [showSpinner,setShowSpinner]=useState(false)
   if (!debugResult?.calldataDetails) return null;
   const { contract, handleCalldata } = debugResult.calldataDetails;
+  
+  const handleClick=async()=>{
+    setButtonText('Simulating');
+    setDisabled(true)
+    setShowSpinner(true)
+    await simulateCall({contract,handleCalldata,chainId,message})
+    setDisabled(false)
+    setShowSpinner(false)
```

Comment:
The disabled state always equals the showSpinner state so there's no reason to create 2 state vars. Can you please use the replace them with a single [loading, setLoading] state that is the inverse of disabled now?